rfcs/text/0000-dss-bundle-enumeration.md

```diff
@@ -0,0 +1,44 @@
+### DCP PR:
+
+***Leave this blank until the RFC is approved** then the **Author(s)** must create a link between the assigned RFC number and this pull request in the format:*
+
+`[dcp-community/rfc#](https://github.com/HumanCellAtlas/dcp-community/pull/<PR#>)`
+
+# RFC: Bundle Enumeration of the DCP Data Store (DSS)
+
+## Summary
+
+This RFC proposes bundle enumeration endpoint(s) for the DSS.
+
+## Author(s)
+
+* [Brian Hannafious](mailto:bhannafi@ucsc.edu)
+
+## Shepherd
+***Leave this blank.** This role is assigned by DCP PM to guide the **Author(s)** through the RFC process.*
+
+*Recommended format for Shepherds:*
+
+ `[Name](mailto:username@example.com)`
+
+## Motivation
+
+Currently, the only means of listing bundles stored in the DSS is the internal Elasticsearch (ES) metadata index, which
+must be kept current with object storage. The DSS should provide bundle enumeration independently of the ES metadata index,
+emphasizing consistency and scalability.
+
+### User Stories
+
+* As a downstream service developer, I would like to enumerate the bundle contents of the DSS so I can create my own
+  index.
+
+* As a downstream service developer, I would like to check if my index contains all the bundles in the DSS.
+
+## Detailed Design
```

Comment:
I was expecting something a bit more [swagger-y](https://dss.data.humancellatlas.org/) rather than a narrative description in **Detailed Design** for a API.

Reply:
@brianraymor It's not clear to me how to address your comment. Perhaps you have something in mind similar to the API endpoint descriptions found in the [Deletion RFC](https://github.com/HumanCellAtlas/dcp-community/blob/0a7cdc63a77581ad891cf6e6bc03bc3046bd8ecc/rfcs/text/0004-dss-deletion-process.md#bundle-deletion-api).

However, those additions will not necessarily improve the clarity and actionability of this document, which defines a simple extension to the DSS API in language that I believe is clear to the developers who will implement it. Do you feel there are technical details missing, or that there is ambiguity that needs to be cleared up?

Reply:
The Deletion RFC more closely meets my expectations for a RFC as _design document_. Another approach is how Azure defines their [REST APIs](https://docs.microsoft.com/en-us/rest/api/). Any reviewer/developer should be able to read this RFC and understand the DSS API in detail - not just the implementers. This currently reads more like what we called _one pagers_ at Microsoft.